test/t8_geometry/t8_geometry_implementations/t8_gtest_geometry_cad.cxx

```diff
@@ -365,6 +367,191 @@ TEST (t8_gtest_geometry_cad, jacobian)
+ protected:
+  void
+  SetUp () override
+  {
+    eclass = GetParam ();
+    T8_ASSERT (0 <= eclass && eclass < T8_ECLASS_COUNT);
+    Handle_Geom_Curve cad_curve;
+    TColgp_Array1OfPnt point_array (1, 3);
+    TopoDS_Shape shape;
+
+    /*  x--> u-parameter
+    *   
+    *                 curve
+    *  ----------------------------------
+    * 
+    *  0               0.5              1
+    */
+
+    point_array (1) = gp_Pnt (0.0, 0.0, 0.0);
+    point_array (2) = gp_Pnt (0.5, 0.0, 0.0);
+    point_array (3) = gp_Pnt (1.0, 0.0, 0.0);
+
+    cad_curve = GeomAPI_PointsToBSpline (point_array).Curve ();
+    shape = BRepBuilderAPI_MakeEdge (cad_curve).Edge ();
+
+    geometry_cad = new t8_geometry_cad (3, shape, "cad curve dim=3");
+
+    geometry = geometry_cad;
+
+    t8_cmesh_init (&cmesh);
+    t8_cmesh_set_tree_class (cmesh, 0, eclass);
+  }
+
+  void
+  TearDown () override
+  {
+    t8_cmesh_destroy (&cmesh);
+  }
+  t8_cmesh_t cmesh;
+  t8_eclass_t eclass;
+  t8_geometry_cad *geometry_cad;
+  t8_geometry_c *geometry;
+  /* The arrays prescribe the linkage of the element. Edge 2 of the element is linked and the face is unlinked. */
+  int faces[1] = { 0 };
+  int edges[8] = { 0, 0, 1, 0, 0, 0, 0, 0 };
+};
+
+TEST_P (class_2d_element_linear_cad_curve, t8_check_2d_element_linear_cad_curve)
+{
+  /* Saving the corner vertices for the given element class. */
+  const int num_vertices = t8_eclass_num_vertices[eclass];
+  const double *vertices = &(t8_element_corner_ref_coords[eclass][0][0]);
+
+  t8_cmesh_set_tree_vertices (cmesh, 0, vertices, num_vertices);
+
+  /* Edges are parametrized in one parameter u. The array contains the parameters
+   * each vertex of the edge has on the linked curve. */
+  double params[2] = { 0, 1 };
+
+  /* Passing of the attributes to the element */
+  t8_cmesh_set_attribute (cmesh, 0, t8_get_package_id (), T8_CMESH_CAD_FACE_ATTRIBUTE_KEY, faces, sizeof (int), 0);
+  t8_cmesh_set_attribute (cmesh, 0, t8_get_package_id (), T8_CMESH_CAD_EDGE_ATTRIBUTE_KEY, edges,
+                          2 * num_vertices * sizeof (int), 0);
+  t8_cmesh_set_attribute (cmesh, 0, t8_get_package_id (), T8_CMESH_CAD_EDGE_PARAMETERS_ATTRIBUTE_KEY + 2, params,
+                          2 * sizeof (double), 0);
+
+  /* Register the geometry */
+  t8_cmesh_register_geometry (cmesh, &geometry);
+  /* Commit the cmesh */
+  t8_cmesh_commit (cmesh, sc_MPI_COMM_WORLD);
+
+  double test_ref_coords[9] = { 0.0, 0.0, 0.0, 0.5, 0.0, 0.0, 1.0, 0.0, 0.0 };
+
+  /* TODO: use randomised test_ref_coords (should lie on linked curve), because the out_coords should be the same, no matter the test_ref_coord. */
+
+  double out_coords[3];
+
+  /* `out_coords` should be equal to the input `ref_coords`. */
+  for (size_t i_coord = 0; i_coord < 3; ++i_coord) {
+    t8_geometry_evaluate (cmesh, 0, test_ref_coords + i_coord * 3, 1, out_coords);
+
+    EXPECT_VEC3_EQ (test_ref_coords + i_coord * 3, out_coords, T8_PRECISION_EPS);
```

Comment:
I am surprised this tolerance works :o

Reply:
The Problem why the first implementation of this Makro didn't work was that we used a `<` instead of a `<=`.